kivydesigner/uix/kdfilechooser.py

```diff
@@ -0,0 +1,461 @@
+        parent_nodes = []
+        if isinstance(node, KDFilechooserEntry):
+            parent_nodes.append(node)
+        else:
+            # By design the top node is not a KDFilechooserEntry
+            # Start at the root children to iterate over all KDFilechooserEntries
+            parent_nodes.extend(self.root.nodes)
+        
+        for parent in parent_nodes:
+            for cur_node in super().iterate_all_nodes(parent):
+                yield cur_node
+
+    def iterate_open_nodes(self, node=None):
+        parent_nodes = []
+        if isinstance(node, KDFilechooserEntry):
+            parent_nodes.append(node)
+        else:
+            # By design the top note is not a KDFilechooserEntry
+            # Start at the root children to iterate over all KDFilechooserEntries
+            parent_nodes.extend(self.root.nodes)
+        
+        for parent in parent_nodes:
+                for cur_node in super().iterate_open_nodes(parent):
+                    yield cur_node
+
+
+class KDFilechooserEntry(BoxLayout, TreeViewNode):
+
+    locked = BooleanProperty(False)
+    '''Locked entries cannot be opened, and are treated as leaf nodes'''
+    path = StringProperty('')
+    '''Full path to the file or directory represented by the current node'''
+    entries = ListProperty([])
+    '''List of child entries'''
+    text = StringProperty('')
+    '''Node display text - set to the file or directory name.'''
+    font_name = StringProperty(DEFAULT_FONT)
+    '''Node display font.'''
+    in_edit_mode = BooleanProperty(False) # MAKE THIS READ ONLY
+    '''Read only properly, set to True if the node text is editable. While in edit
+    mode the entry displays an editable TextInput widget. Only the selected node 
+    is able to be edited, so only one node can be edited at a time.'''
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self._set_text_viewer(False)
+
+    def on_text(self, instance, new_text):
+        self._text_viewer.text = new_text
+
+    def on_touch_down(self, touch):
+        if self.in_edit_mode:
+            self._text_viewer.on_touch_down(touch)
+
+    def on_is_selected(self, instance, is_sel):
+        # Disable edit mode if the node is de-selected
+        if self.in_edit_mode and (not is_sel):
+            self.disable_edit_mode()
+
+    def disable_edit_mode(self):
+        self._set_text_viewer(False)
+
+    def enable_edit_mode(self):
+        self._set_text_viewer(True)
+
+    def get_focus_widget(self):
+        '''Return the focusable widget. Return None if the node is in read only mode.'''
+        if self.in_edit_mode:
+            return self._text_viewer
+        return None
+
+    def on_width(self, instance, new_width):
+        self._text_viewer.text_size = new_width, None
+
+    def on_font_name(self, *args):
+        self._text_viewer.font_name = self.font_name
+
+    def _set_text_viewer(self, edit_mode):
+        # Add code to handle text input value here. 
```

Comment:
Delete obsolete comment.

Reply:
Fixed in [803bdac](https://github.com/Joshua-Douglas/kivydesigner/pull/17/commits/803bdac05c273f7db0d0e9b4a6e426af4bf42305).